extensions/trace_propagators/src/test/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagatorTest.java

```diff
@@ -313,6 +362,78 @@ void extract_UrlEncodedContext() throws UnsupportedEncodingException {
                 TRACE_ID, SPAN_ID, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT));
   }
 
+  @Test
+  void extract_SampledContext_withBaggage() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    JaegerSpanContext context =
+        new JaegerSpanContext(
+            TRACE_ID_HI, TRACE_ID_LOW, SPAN_ID_LONG, DEPRECATED_PARENT_SPAN_LONG, (byte) 5);
+    carrier.put(PROPAGATION_HEADER, TextMapCodec.contextAsString(context));
+    carrier.put(BAGGAGE_PREFIX + "foo", "bar");
+
+    assertThat(getSpanContext(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(
+            SpanContext.createFromRemoteParent(
+                TRACE_ID, SPAN_ID, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT));
+    assertThat(getBaggage(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(Baggage.builder().put("foo", "bar").build());
+  }
+
+  @Test
+  void extract_baggageOnly_withPrefix() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(BAGGAGE_PREFIX + "nometa", "nometa-value");
+    carrier.put(BAGGAGE_PREFIX + "meta", "meta-value");
+    carrier.put("another", "value");
+
+    assertThat(getBaggage(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(
+            Baggage.builder().put("nometa", "nometa-value").put("meta", "meta-value").build());
+  }
+
+  @Test
+  void extract_baggageOnly_withPrefix_emptyKey() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(BAGGAGE_PREFIX, "value"); // Not really a valid key.
+
+    assertThat(getBaggage(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(Baggage.empty());
+  }
+
+  @Test
+  void extract_baggageOnly_withHeader() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(BAGGAGE_HEADER, "nometa=nometa-value,meta=meta-value");
+
+    assertThat(getBaggage(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(
+            Baggage.builder().put("nometa", "nometa-value").put("meta", "meta-value").build());
+  }
+
+  @Test
+  void extract_baggageOnly_withHeader_invalid() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(BAGGAGE_HEADER, "nometa+novalue");
+
+    assertThat(getBaggage(jaegerPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(Baggage.empty());
+  }
+
+  @Test
+  void extract_baggageOnly_withHeader_andPrefix() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(BAGGAGE_HEADER, "nometa=nometa-value,meta=meta-value");
```

Comment:
Can you add some test cases with invalid, and with whitespace?

Reply:
Oh, yes, totally missed that.